app/src/main/java/org/gateshipone/odyssey/activities/OdysseyMainActivity.java

```diff
@@ -142,6 +148,10 @@ protected void onCreate(Bundle savedInstanceState) {
                     if (extras != null && extras.getString(MAINACTIVITY_INTENT_EXTRA_REQUESTEDVIEW, "").equals(MAINACTIVITY_INTENT_EXTRA_REQUESTEDVIEW_NOWPLAYINGVIEW)) {
                         mShowNPV = true;
                     }
+
+                    if (extras != null && extras.getString(MAINACTIVITY_INTENT_EXTRA_BACKTOSETTINGS, "").equals(MAINACTIVITY_INTENT_EXTRA_BACKTOSETTINGS_VALUE)) {
```

Comment:
I think it would be better if you merge both if statements, so that you don't check the extras for null twice.